### PR TITLE
LCAM-1628-schema changes 

### PR DESCRIPTION
**Pull request by loganathanMOJ**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LCAM-1628)

Changes made to use new schema for crown court outcomes

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] GitHub should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

## Additional checks

- Don’t forget to [run](https://github.com/ministryofjustice/laa-crimeapps-maat-functional-tests/actions/workflows/ExecuteUiTests.yaml) the MAAT functional test suite after deploying your changes to the DEV or TEST environments to ensure your changes haven’t broken any of the functional tests.

### CODE REVIEW

**Review by Mhowell494 — CHANGES_REQUESTED** [review]
The `apiUpdateCrownCourtRequest` schema should extend `updateApplicationRequest`. This will allow us to remove many duplicate fields from `apiUpdateCrownCourtRequest`.

### Proposed Changes
1. Move the following fields from `apiUpdateApplicationRequest` to `apiUpdateCrownCourtRequest`:
   - `emstCode`
   - `incomeEvidenceReceivedDate`
   - `capitalEvidenceReceivedDate`
   - `capitalEvidence`

2. In the `apiProcessRepOrderRequest` schema:
   - Financial assessment and passport assessment fields should be made nullable.
   - Update the `RepOrderService.getRepDecision()` method to include additional null checks to accommodate these changes.

3. Update the mapping logic in the orchestration service to reflect the changes outlined above.

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/builder/CrownCourtDTOBuilder.java [hunk 1]
```diff
@@ -18,7 +18,16 @@
 public class CrownCourtDTOBuilder {
 
     public static CrownCourtDTO build(final ApiProcessRepOrderRequest request) {
-        CrownCourtDTO.CrownCourtDTOBuilder builder = CrownCourtDTO.builder()
+        CrownCourtDTO.CrownCourtDTOBuilder builder = mapToCrownCourtDTO(request);
+
+        if (request instanceof ApiUpdateApplicationRequest updateRequest) {
+            mapUpdateApplicationRequestToCrownCourtDTO(updateRequest, builder);
```
Comment (Mhowell494) [inline crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/builder/CrownCourtDTOBuilder.java:24]: Can't we just add another condition here to build the `UpdateCrownCourt` specific fields?

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownProceedingControllerTest.java [hunk 2]
```diff
@@ -157,9 +157,9 @@ void givenAInValidInput_whenUpdateIsInvoked_RequestObjectFailsValidation() throw
...
     void givenAnInput_whenUpdateIsInvokedAndValidationFails_thenBadRequestResponseIsReturned() throws Exception {
-        var apiUpdateApplicationRequest =
-                TestModelDataBuilder.getApiUpdateApplicationRequest(IS_VALID);
-        var updateApplicationRequestJson = objectMapper.writeValueAsString(apiUpdateApplicationRequest);
+        var apiUpdateCrownCourtRequest =
+                TestModelDataBuilder.getApiUpdateCrownCourtRequest(IS_VALID);
+        var updateApplicationRequestJson = objectMapper.writeValueAsString(apiUpdateCrownCourtRequest);
         when(crownCourtDetailsValidator.checkCCDetails(any())).thenThrow(new ValidationException("Cannot have Crown Court outcome without Mags Court outcome"));
         mvc.perform(RequestBuilderUtils.buildRequestGivenContent(
                         HttpMethod.PUT, updateApplicationRequestJson, ENDPOINT_URL + "/update-crown-court"))
```
Comment (Mhowell494) [inline crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownProceedingControllerTest.java:165]: Can we make this a constant? I also think we should be reusing the error message constant from the `CrownCourtDetailsValidator`

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownProceedingControllerTest.java [hunk 3]
```diff
@@ -131,9 +131,9 @@ void updateApplication_BadRequest_RequestEmptyBody() throws Exception {
 
     @Test
     void givenAValidInput_whenUpdateIsInvoked_thenSuccess() throws Exception {
```
Comment (Mhowell494) [inline crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownProceedingControllerTest.java:133]: Test names are a bit unclear, can we update these to reflect we're targeting the `updateCrownCourt` endpoint?

File: crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownProceedingControllerTest.java [hunk 4]
```diff
@@ -146,9 +146,9 @@ void givenAValidInput_whenUpdateIsInvoked_thenSuccess() throws Exception {
 
     @Test
     void givenAInValidInput_whenUpdateIsInvoked_RequestObjectFailsValidation() throws Exception {
-        var apiUpdateApplicationRequest =
```
Comment (Mhowell494) [inline crown-court-proceeding/src/test/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownProceedingControllerTest.java:149]: The other endpoints also have tests to check for a missing or empty request body, can we add these in please?

**Review by Mhowell494 — CHANGES_REQUESTED** [review]

**Comment by Mhowell494** [comment]
There doesn't appear to be any changes to the business logic in the `RepOrderService`. Now that the financial/passport assessment fields are nullable we need to add in some additional checks to prevent possible `NullPointerExceptions`.

File: crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownProceedingApi.java [hunk 5]
```diff
@@ -55,8 +55,8 @@ ResponseEntity<ApiUpdateApplicationResponse> updateApplication(
             )
     )
     @DefaultHTTPErrorResponse
-    ResponseEntity<ApiUpdateCrownCourtOutcomeResponse> update(
-            @Parameter(description = "Updated Application Data",
+    ResponseEntity<ApiUpdateCrownCourtOutcomeResponse> updateCrownCourt(
+            @Parameter(description = "Updated crown court",
```
Comment (Mhowell494) [inline crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/controller/CrownProceedingApi.java:59]: I would probably change this to `"Updated Crown Court Data"`

**Review by Mhowell494 — APPROVED** [review]
